### ISSUE TITLE
Build fails: undefined reference to main

### ISSUE THREAD

**Issue by otheus** [jmkeyes/adsmpipe#3]
Executing `make -d VERBOSE=1 SHELL='sh -x' PLATFORM=linux  all`

```
Live child 0x0133ffd0 (source/adsmpipe.o) PID 3770
+ echo '    CC source/adsmpipe.o'
    CC source/adsmpipe.o
Reaping winning child 0x0133ffd0 PID 3770
Live child 0x0133ffd0 (source/adsmpipe.o) PID 3771
+ cc -m64 -I/opt/tivoli/tsm/client/api/bin64/sample/ -c source/adsmpipe.c -o source/adsmpipe.o
Reaping winning child 0x0133ffd0 PID 3771
Removing child 0x0133ffd0 PID 3771 from chain.
    Successfully remade target file `source/adsmpipe.o'.
    Considering target file `source/adsmblib.o'.
     File `source/adsmblib.o' does not exist.
      Pruning file `source/adsmpipe.c'.
      Pruning file `source/adsmblib.c'.
     Finished prerequisites of target file `source/adsmblib.o'.
    Must remake target `source/adsmblib.o'.
Putting child 0x0133ffb0 (source/adsmblib.o) PID 3774 on the chain.
Live child 0x0133ffb0 (source/adsmblib.o) PID 3774
+ echo '    CC source/adsmblib.o'
    CC source/adsmblib.o
Reaping winning child 0x0133ffb0 PID 3774
Live child 0x0133ffb0 (source/adsmblib.o) PID 3775
+ cc -m64 -I/opt/tivoli/tsm/client/api/bin64/sample/ -c source/adsmblib.c -o source/adsmblib.o
Reaping winning child 0x0133ffb0 PID 3775
Removing child 0x0133ffb0 PID 3775 from chain.
    Successfully remade target file `source/adsmblib.o'.
   Finished prerequisites of target file `adsmpipe'.
  Must remake target `adsmpipe'.
Putting child 0x0133ca60 (adsmpipe) PID 3778 on the chain.
Live child 0x0133ca60 (adsmpipe) PID 3778
+ echo '    LD adsmpipe'
    LD adsmpipe
Reaping winning child 0x0133ca60 PID 3778
Live child 0x0133ca60 (adsmpipe) PID 3779
+ cc -m64 -I/opt/tivoli/tsm/client/api/bin64/sample/ -L/opt/tivoli/tsm/client/api/bin64/ -lApiTSM64 -o adsmpipe source/adsmpipe.o source/adsmblib.o
/usr/lib/gcc/x86_64-redhat-linux/4.4.7/../../../../lib64/crt1.o: In function `_start':
(.text+0x20): undefined reference to `main'
collect2: ld returned 1 exit status
Reaping losing child 0x0133ca60 PID 3779
make[1]: *** [adsmpipe] Error 1
```

**Comment by otheus** [comment]
Source files were actually empty:

```
$ ls -l source/
total 8
-rw-r--r--. 1 xxx xxx 0 Apr 20 19:24 adsmblib.c
-rw-r--r--. 1 xxx xxx 940 Apr 20 19:25 adsmblib.o
-rw-r--r--. 1 xxx xxx 0 Apr 20 19:24 adsmpipe.c
-rw-r--r--. 1 xxx xxx 940 Apr 20 19:25 adsmpipe.o
```

**Comment by otheus** [comment]
Project requires `ncompress` on RedHat, which provides Z utilities used by tar. Consider replacing `tar -Z` with explicity commands to compress/decompress.

**Comment by jmkeyes** [comment]
Indeed. On RedHat platforms `ncompress` is required to extract the `adsmpipe` source code from IBM's original archive. I recently modified the build process to download the original source code from IBM's FTP server (and extract it automatically) instead of bundling the source code itself. I'll make a note in the README.

**Comment by jmkeyes** [comment]
Fixed as of 298d2410943becbf1cf9e74113d4773e505e5615.
